src/Nethermind/Nethermind.Evm/TransactionProcessing/TransactionProcessor.cs

```diff
@@ -166,37 +192,40 @@ private void Execute(Transaction transaction, BlockHeader block, ITxTracer txTra
                 }
             }
 
-            UInt256 senderReservedGasPayment = isCall ? UInt256.Zero : (ulong) gasLimit * gasPrice;
+            UInt256 senderReservedGasPayment = restore ? UInt256.Zero : (ulong) gasLimit * gasPrice;
```

Comment:
how about !commit?

Reply:
we still modify the state, its as intended